cmake/libvpx.cmake

```diff
@@ -694,31 +701,68 @@ elseif (is_arm OR is_aarch64)
 
     # 32-bit assembly with NEON instructions
     if (arm_use_neon AND (NOT is_aarch64))
+
+        # Convert the ASM files from the ARM Developer Suite 1.0.1 syntax
+        # to syntax accepted by the C compiler's assembler.
+
+        string(REGEX REPLACE
+            "${CMAKE_CURRENT_SOURCE_DIR}" "${CMAKE_CURRENT_BINARY_DIR}"
+            libvpx_loc_build ${libvpx_loc})
+
+        list(APPEND include_directories
+            ${libvpx_loc_build}/source/libvpx
+        )
+
+        set(libvpx_arm32_asm_dir_src ${libvpx_loc}/source/libvpx/vpx_dsp/arm)
+        set(libvpx_arm32_asm_dir_obj ${libvpx_loc_build}/source/libvpx/vpx_dsp/arm)
+
+        file(MAKE_DIRECTORY ${libvpx_arm32_asm_dir_obj})
+
+        set(libvpx_arm32_asm_in
+            intrapred_neon_asm.asm
+            vpx_convolve_copy_neon_asm.asm
+            vpx_convolve8_horiz_filter_type2_neon.asm
+            vpx_convolve8_vert_filter_type2_neon.asm
+            vpx_convolve8_horiz_filter_type1_neon.asm
+            vpx_convolve8_vert_filter_type1_neon.asm
+            vpx_convolve8_avg_horiz_filter_type2_neon.asm
+            vpx_convolve8_avg_vert_filter_type2_neon.asm
+            vpx_convolve8_avg_horiz_filter_type1_neon.asm
+            vpx_convolve8_avg_vert_filter_type1_neon.asm
+            vpx_convolve_avg_neon_asm.asm
+            loopfilter_16_neon.asm
+            loopfilter_8_neon.asm
+            loopfilter_4_neon.asm
+            save_reg_neon.asm
+            idct_neon.asm
+            idct4x4_1_add_neon.asm
+            idct4x4_add_neon.asm)
+
+        foreach(in_file ${libvpx_arm32_asm_in})
+            set(out_file ${in_file}.S)
+
+            set(in_file_path  ${libvpx_arm32_asm_dir_src}/${in_file})
+            set(out_file_path ${libvpx_arm32_asm_dir_obj}/${out_file})
+
+            add_custom_command(
+                OUTPUT ${out_file_path}
+                COMMAND sh -c \"cat \'${in_file_path}\' | ${PERL_EXECUTABLE} \'${ads2gas}\' > \'${out_file_path}\'\"
+                DEPENDS ${in_file_path}
+            )
+
+            nice_target_sources(libvpx ${libvpx_arm32_asm_dir_obj}
+            PRIVATE
+                ${out_file}
+            )
+        endforeach()
```

Comment:
Probabluy the right way is to create a target_ads2gas_sources, just like target_yasm_source and give it .asm files

Reply:
Thanks for the suggestion, I'll try to re-factor it on the following weekend if I'll have time

Reply:
Finally found some free time, and agreed, it looks much better now.  (my experience with cmake is rather limited)

I put it into the libvpx.cmake file, since the ads2gas scripts are specific to libvpx, but if you feel like it should be in a separate file, I guess we could move it.  (it would probably need to be made more generic then, with parameters for a script file and such)

Reply:
I think it's fine. Gladly, we have #55 also now. Thank you very much.

Reply:
You're welcome, I'm sorry it took so long.